flink-runtime/src/main/java/org/apache/flink/runtime/metrics/TimerGauge.java

```diff
@@ -32,9 +32,21 @@
  * happen in a couple of hours, the returned value will account for this ongoing measurement.
  */
 public class TimerGauge implements Gauge<Long>, View {
+
+    private static final int DEFAULT_TIME_SPAN_IN_SECONDS = 60;
+
     private final Clock clock;
 
-    private long previousCount;
+    /** The time-span over which the average is calculated. */
+    private final int timeSpanInSeconds;
+    /** Circular array containing the history of values. */
+    private final long[] values;
+    /** The index in the array for the current time. */
+    private int time = 0;
```

Comment:
I think `idx` would be a better name for this. Time is confusing because this actually gets reset to zero when the array is full.